### PR TITLE
optimisation to limit out of memory when loading a lot of big internet picture

### PR DESCRIPTION
**Pull request by alain57**
When using this helper for a lot of pictures that are not mobile
optimized won get an out of memory very fast.
The trick is to display a scaled down image.

By default, images have a MAX_DISPLAY_SIZE of 100
so if you load an image that is 2000x2000 pixel big, it will be shown
as a 100x100 picture.

The setUrlDrawable methods also offer the ability to set an other value
if desired

### CODE REVIEW

File: UrlImageViewHelper/src/com/koushikdutta/urlimageviewhelper/UrlImageViewHelper.java [hunk 1]
```diff
@@ -58,14 +58,40 @@ private static void prepareResources(Context context) {
...
-    private static BitmapDrawable loadDrawableFromStream(Context context, InputStream stream) {
+    private static BitmapDrawable loadDrawableFromStream(Context context, String filename, int maxSize) throws IOException
+    {
+        FileInputStream  stream = context.openFileInput(filename);
         prepareResources(context);
-        final Bitmap bitmap = BitmapFactory.decodeStream(stream);
+
+        BitmapFactory.Options o = new BitmapFactory.Options();
+        o.inJustDecodeBounds = true;
+        BitmapFactory.decodeStream(stream,null,o);
```
Comment (koush) [inline UrlImageViewHelper/src/com/koushikdutta/urlimageviewhelper/UrlImageViewHelper.java:68]: Looks good! Mind cleaning up the formatting so it matches the rest of the file? Spaces between operators, arguments, etc? Thanks!

**Comment by koush** [comment]
Looks good! Just one comment regarding formatting :)

**Comment by koush** [comment]
Also I just realized that this is limiting everything using default args to a max size of 100. Most devices are at least 480 pixels across now.
I would say by default, the sampling size stuff should be disabled. Applications recompiling against a new version of this library may experience unintended effects.

**Comment by koush** [comment]
Perhaps a better way to do this would be to have a callback that returns a BitmapFactory.Options to use for decoding?

**Comment by alain57** [comment]
sorry for the 100 value, i set it because i use your code to set thumbnails on the app i'm coding ... 
and also sorry for the bad formatting (came from copy-paste on other online code)... i put attention to the modification i'd done, but forget to check that part :( sorry

for the callback... to said the truth i have no really idea how and where to set it... (its my first android project :'( )

**Comment by alain57** [comment]
correct the formating and the default value

**Comment by koush** [comment]
What I meant by the default value, is that by default, the library should not be changing the sampling size at all (not that it should be 480).

**Comment by alain57** [comment]
maybe it would be best if the default sampling value, is the maximum available screen size, so this don't affect the most users ?

Or should it really be without sampling at all ?

**Comment by koush** [comment]
No sampling at all. Default behavior should not tamper with the original image.

**Comment by ChristopheVersieux** [comment]
This one could have been interesting...
I am thinking for instance, to display a downscaled picture in a ListView.

I am currently getting a lot of OutOfMemoryErrors because it seems that the listview load the full picture.

Maybe I am wrong, but a parameter in the setUrldDrawable to specify the inSampleSize would be a nice addition, so people could use it in ListView

**Pull request closed without merging**